src/fluree/db/util/log.cljc

```diff
@@ -135,4 +135,27 @@
      [msg c]
      `(debug-async->vals ~c ~msg)))
 
+#?(:clj
+   (defn with-mdc* [context f]
+     (let [original (org.slf4j.MDC/getCopyOfContextMap)]
+       (try
+         (doseq [[k v] context]
+           (org.slf4j.MDC/put (name k) (str v)))
+         (f)
+         (finally
+           (org.slf4j.MDC/clear)
+           (when original
+             (doseq [[k v] original]
+               (org.slf4j.MDC/put k v)))))))
+
+   :cljs
+   (defn with-mdc* [_context f]
+     (f)))
+
+#?(:clj
+   (defmacro with-mdc [context & body]
```

Comment:
Could you post an example of how this should be used?

Reply:
https://github.com/fluree/nexus-server/pull/3/files#diff-fe046dc3eec2629e801daaa120b65f8049c49cc18ec59481e48ee8c23a89d1acR17

Reply:
In this example, does the `:ledger.id` and `:tx.id` values persist from that single call, or do we have to repeat it?

If we did this:
```clojure
(log/with-mdc {:ledger.id ledger-id :tx.id tx-id}
  (log/debug "begin ..."))

(log/debug "continuing ...")

(log/debug "end ...")
```
then will the `:ledger.id` and `:tx.id` values show up in the "continuing ..." and "end ..." logs?

Reply:
No.  It works more like binding when the with-mdc ends the values are removed. Since they are associated with the thread this ensures they get removed when the context such as a request is finished.

Reply:
sorry for losing track of this. So `log/with-mdc` is really meant to wrap all the top level api calls, and then they will automatically add the context to any log calls from any other functions within the call stack. Makes sense.

We probably want to wrap the functions in `fluree.db.api` with `log/with-mdc`, but that can come later.